Android/app/src/main/java/org/opendroneid/android/bluetooth/WiFiBeaconScanner.java

```diff
@@ -0,0 +1,208 @@
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+
+import org.opendroneid.android.log.LogMessageEntry;
+import org.opendroneid.android.log.LogWriter;
+
+public class WiFiBeaconScanner {
+  private static final int OUILen = 3;
+  private static final int DriStartByteOffset = 4;
+  private static final int ScanTimerInterval = 5;
+  private static final int DRIOUI[] = {0x90, 0x3A, 0xE6};
+  private boolean WiFiScanEnabled = true;
+  private final OpenDroneIdDataManager dataManager;
+  private final LogWriter logger;
+  private WifiManager wifiManager;
+  private Handler handler;
+  Context context;
+  int scanSuccess;
+  int scanFailed;
+  String startTime;
+  CountDownTimer countDownTimer;
+  boolean beaconScanDebugEnable;
+
+  private static final String TAG = WiFiBeaconScanner.class.getSimpleName();
+
+  @RequiresApi(api = Build.VERSION_CODES.O)
+  public WiFiBeaconScanner(Context context, OpenDroneIdDataManager dataManager, LogWriter logger) {
+    this.dataManager = dataManager;
+    this.logger = logger;
+
+    this.startTime = getCurrTimeStr();
+
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.R ||
+        !context.getPackageManager().hasSystemFeature(PackageManager.FEATURE_WIFI)) {
+      Toast.makeText(context, "WiFi Scanning is not supported", Toast.LENGTH_LONG).show();
+      WiFiScanEnabled = false;
+      return;
+    }
+    if (!WiFiScanEnabled) {
+      Toast.makeText(context, "WiFi Scanning is disabled", Toast.LENGTH_LONG).show();
+      return;
+    }
+
+    this.context = context;
+    handler = new Handler();
+    beaconScanDebugEnable = false;
+
+    wifiManager = (WifiManager)context.getSystemService(Context.WIFI_SERVICE);
+    if (!wifiManager.isWifiEnabled()) {
+      Toast.makeText(context, "Turning WiFi ON...", Toast.LENGTH_LONG).show();
+      wifiManager.setWifiEnabled(true);
+    }
+    IntentFilter filter = new IntentFilter(WifiManager.SCAN_RESULTS_AVAILABLE_ACTION);
+    BroadcastReceiver myReceiver = new BroadcastReceiver() {
+      @Override
+      public void onReceive(Context context, Intent intent) {
+        handleScanResults(intent);
+      }
+    };
+    context.registerReceiver(myReceiver, filter);
+
+    startCountDownTimer(context);
+    // Kick off WiFi Scan
+    startScan();
+
+  }
+
+  void processRemoteIdVendorIE(ScanResult scanResult, ScanResult.InformationElement element) {
+    ByteBuffer buf = element.getBytes();
+    byte[] driOUI = new byte[OUILen];
+    byte[] arr = new byte[buf.remaining()];
+    buf.get(driOUI, 0, OUILen);
+    if ((driOUI[0] & 0xFF) == DRIOUI[0] && (driOUI[1] & 0xFF) == DRIOUI[1] &&
+            (driOUI[2] & 0xFF) == DRIOUI[2]) {
+      buf.position(DriStartByteOffset);
+      buf.get(arr, 0, buf.remaining());
+      LogMessageEntry logMessageEntry = new LogMessageEntry();
+      long timeNano = SystemClock.elapsedRealtimeNanos();
+      dataManager.receiveDataWiFi(arr, scanResult.BSSID, scanResult.BSSID.hashCode(),
```

Comment:
It would be useful to pass also the scanResult.level (the RSSI value) to the dataManager, so it can be displayed on the UI.